BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -2030,5 +1893,57 @@ private float AdjustValue(float min, float max, float value, float step)
 
             return adjusted_value;
         }
+
+        private Device GetDevice()
+        {
+            Device device = _context.Devices[0];
+
+            if (string.IsNullOrEmpty(this.SerialNumber))
+                return device;
+
+            foreach(Device dev in _context.Devices)
+            {
+                if (dev.Info[CameraInfo.SerialNumber] == this.SerialNumber)
+                {
+                    device = dev;
+                    break;
+                }
+            }
+
+            return device;
+        }
+
+        private Sensor GetSensor(string sensorName)
+        {
+            Device dev = GetDevice();
+            Sensor s = dev.Sensors[0];
+
+            foreach(Sensor sen in dev.Sensors)
+            {
+                Sensor.CameraInfos infos = sen.Info;
+                string info = infos[CameraInfo.Name];
+                if(info == sensorName)
+                {
+                    s = sen;
+                    break;
+                }
+            }
+
+            return s;
```

Comment:
I would throw an Exception if the sensorName is invalid, instead of returning the first sensor.

Reply:
Okay, I'll do the same thing with the stream profile and the device if a S/N is specified but the device could not be found.